mongodb/mongo_dataset.cc

```diff
@@ -0,0 +1,267 @@
+
+#include "bsoncxx/oid.hpp"
+#include "bsoncxx/builder/stream/document.hpp"
+#include "bsoncxx/builder/stream/array.hpp"
+#include "mongocxx/client.hpp"
+
+#include "mldb/core/function.h"
+#include "mldb/core/dataset.h"
+#include "mldb/types/structure_description.h"
+#include "mldb/rest/rest_request_router.h"
+#include "mldb/types/any_impl.h"
+
+#include "mongo_common.h"
+
+using namespace std;
+
+namespace Datacratic {
+namespace MLDB {
+namespace Mongo {
+
+typedef tuple<Path, CellValue, Date> Cell;
+
+struct MongoMatrixView : MatrixView {
+
+    vector<Path> getRowNames(ssize_t start = 0,
+                             ssize_t limit = -1) const override
+    {
+        return vector<Path>{};
+    }
+
+    vector<RowHash> getRowHashes(ssize_t start = 0,
+                                 ssize_t limit = -1) const override
+    {
+        return vector<RowHash>{};
+    }
+
+    size_t getRowCount() const override
+    {
+        return -1;
+    }
+
+    bool knownRow(const Path & rowName) const override
+    {
+        return false;
+    }
+    
+    MatrixNamedRow getRow(const Path & rowName) const override
+    {
+        throw HttpReturnException(500, "Unimplemented getRow");
+    }
+
+    Path getRowName(const RowHash & row) const override
+    {
+        throw HttpReturnException(500, "Unimplemented getRowName");
+    }
+
+    bool knownColumn(const Path & columnName) const override
+    {
+        return false;
+    }
+    
+    Path getColumnName(ColumnHash column) const override
+    {
+        throw HttpReturnException(500, "Unimplemented getColumnName");
+    }
+
+    vector<Path> getColumnNames() const override
+    {
+        return vector<Path>{};
+    }
+
+    size_t getColumnCount() const override
+    {
+        return -1;
+    }
+
+    uint64_t getRowColumnCount(const Path & row) const override
+    {
+        return 0;
+    }
+};
+
+struct MongoColumnIndex : ColumnIndex {
+
+    typedef std::function<bool (const Path & columnName,
+                                const ColumnStats & stats)> OnColumnStats;
+
+    MatrixColumn getColumn(const Path & column) const override
+    {
+        throw HttpReturnException(500, "Unimplemented getColumn");
+    }
+
+    bool knownColumn(const Path & column) const override
+    {
+        return false;
+    }
+
+    vector<Path> getColumnNames() const override
+    {
+        return vector<Path>{};
+    }
+
+    vector<Path> getRowNames(ssize_t start = 0,
+                             ssize_t limit = -1) const override
+    {
+        return vector<Path>{};
+    }
+};
+
+struct MongoDatasetConfig {
+    static constexpr const char * name = "mongodb.dataset";
+    MongoDatasetConfig() {}
+    string connectionScheme;
+    string collection;
+};
+
+DECLARE_STRUCTURE_DESCRIPTION(MongoDatasetConfig);
+DEFINE_STRUCTURE_DESCRIPTION(MongoDatasetConfig);
+
+MongoDatasetConfigDescription::
+MongoDatasetConfigDescription()
+{
+    addField("connectionScheme", &MongoDatasetConfig::connectionScheme,
```

Comment:
Unless this the standard naming in Mongo's docs.  I would rename to `connectionUri`.  The scheme is really just the first part of the uri.